### PR TITLE
Adding basic_info and connection validations

### PR DESCRIPTION
**Pull request by pro-akim**
Adding granularity to tests

It was included:

- [x] basic_info
- [x] connection

Some additional validation in arch, installation os version, port and processes were added

### CODE REVIEW

**Comment by pro-akim** [comment]
## Update

TEST

<details><summary>YAML file</summary>

```
version: 0.1
description: This workflow is used to test agents' deployment for DDT1 PoC
variables:
  agent-os:
    - linux-oracle-9-amd64
    - linux-centos-7-amd64
    - linux-centos-8-amd64
    - linux-redhat-7-amd64
    - linux-redhat-8-amd64
    - linux-redhat-9-amd64
    - linux-amazon-2-amd64
  manager-os: linux-ubuntu-22.04-amd64
  infra-provider: vagrant
  working-dir: /tmp/dtt1-poc

tasks:
  # Unique manager allocate task
  - task: "allocate-manager-{manager-os}"
    description: "Allocate resources for the manager."
    do:
      this: process
      with:
        path: python3
        args:
          - modules/allocation/main.py
          - action: create
          - provider: "{infra-provider}"
          - size: large
          - composite-name: "{manager-os}"
          - inventory-output: "{working-dir}/manager-{manager-os}/inventory.yaml"
          - track-output: "{working-dir}/manager-{manager-os}/track.yaml"
          - label-termination-date: "1d"
          - label-team: "qa"
    on-error: "abort-all"
    cleanup:
      this: process
      with:
        path: python3
        args:
          - modules/allocation/main.py
          - action: delete
          - track-output: "{working-dir}/manager-{manager-os}/track.yaml"

  # Unique agent allocate task
  - task: "allocate-agent-{agent}"
    description: "Allocate resources for the agent."
    do:
      this: process
      with:
        path: python3
        args:
          - modules/allocation/main.py
          - action: create
          - provider: "{infra-provider}"
          - size: small
          - composite-name: "{agent}"
          - inventory-output: "{working-dir}/agent-{agent}/inventory.yaml"
          - track-output: "{working-dir}/agent-{agent}/track.yaml"
          - label-termination-date: "1d"
          - label-team: "qa"
    foreach:
      - variable: agent-os
        as: agent
    cleanup:
      this: process
      with:
        path: python3
        args:
          - modules/allocation/main.py
          - action: delete
          - track-output: "{working-dir}/agent-{agent}/track.yaml"
    depends-on:
      - "provision-manager-{manager-os}"

  # Unique manager provision task
  - task: "provision-manager-{manager-os}"
    description: "Provision the manager."
    do:
      this: process
      with:
        path: python3
        args:
          - modules/provision/main.py
          - inventory: "{working-dir}/manager-{manager-os}/inventory.yaml"
          - install:
            - component: wazuh-manager
              type: assistant
              version: 4.7.3
              live: True
    depends-on:
      - "allocate-manager-{manager-os}"


  # Generic agent test task
  - task: "run-agent-{agent}-tests"
    description: "Run tests install for the agent {agent}."
    do:
      this: process
      with:
        path: python3
        args:
          - modules/testing/main.py
          - targets:
            - wazuh-1: "{working-dir}/manager-{manager-os}/inventory.yaml"
            - agent: "{working-dir}/agent-{agent}/inventory.yaml"
          - tests: "install,registration,connection,basic_info,restart,stop,uninstall"
          - component: "agent"
          - wazuh-version: "4.7.3"
          - wazuh-revision: "40714"
          - live: "True"
    foreach:
      - variable: agent-os
        as: agent
    depends-on:
      - "allocate-agent-{agent}"

```
</details>

File: deployability/modules/testing/tests/test_agent/test_basic_info.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+import pytest
+import re
+
+from ..helpers.agent import WazuhAgent, WazuhAPI
+from ..helpers.constants import WAZUH_ROOT
+from ..helpers.generic import HostConfiguration, HostInformation, GeneralComponentActions, Waits
+from ..helpers.logger.logger import logger
```
Comment (fcaffieri) [inline deployability/modules/testing/tests/test_agent/test_basic_info.py:11]:
```suggestion
from modules.testing.utils import logger
```

We need to implement logger with the new fixes

File: deployability/modules/testing/tests/test_agent/test_connection.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+import pytest
+import re
+
+from ..helpers.agent import WazuhAgent, WazuhAPI
+from ..helpers.generic import HostInformation, GeneralComponentActions, Waits
+from ..helpers.manager import WazuhManager, WazuhAPI
+from ..helpers.logger.logger import logger
```
Comment (fcaffieri) [inline deployability/modules/testing/tests/test_agent/test_connection.py:11]:
```suggestion
from modules.testing.utils import logger
```

Same

**Review by fcaffieri — CHANGES_REQUESTED** [review]
GJ, some minor fixes

File: deployability/modules/testing/tests/test_agent/test_basic_info.py (hunk 1 above)
Comment (pro-akim) [reply deployability/modules/testing/tests/test_agent/test_basic_info.py]: Done https://github.com/wazuh/wazuh-qa/pull/5225/commits/3220fb711cbfc202b9591d26d10576732b8cd2e8

**Pull request closed without merging**